packages/workspace/src/browser/workspace-storage-service.ts

```diff
@@ -33,12 +34,11 @@ export class WorkspaceStorageService implements StorageService {
 
     @postConstruct()
     protected init() {
-        this.initialized = this.workspaceService.root.then(stat => {
-            if (stat) {
-                this.prefix = stat.uri;
-            } else {
-                this.prefix = '_global_';
-            }
+        this.initialized = this.workspaceService.workspace.then(data => {
+            this.prefix = this.getPrefix(data.workspaceFolder);
+        });
+        this.workspaceService.onWorkspaceChanged(event => {
+            this.prefix = this.getPrefix(event.workspaceFolder);
```

Comment:
How could it happen that `event.workspaceFolder` is changed? Won't we lose the storage data for the current page by switching a prefix? Is it an effect of mixing a project and workspace in one entity?

Reply:
You are right. the workspaceFolder never changes unless theia reloads, and therefore we don't need to watch the `onWorkspaceChanged` here.